genmethod.go

```diff
@@ -231,7 +238,12 @@ func (m *marshalMethod) marshalConversions(from, to Var, format string) (s []Sta
 	for _, f := range m.mtyp.Fields {
 		accessFrom := Dotted{Receiver: from, Name: f.name}
 		accessTo := Dotted{Receiver: to, Name: f.name}
-		conversion := m.convert(accessFrom, accessTo, f.origTyp, f.typ)
+		var conversion []Statement
+		if f.function != nil {
+			conversion = m.convert(CallFunction{Func: accessFrom}, accessTo, f.origTyp, f.typ)
+		} else {
+			conversion = m.convert(accessFrom, accessTo, f.origTyp, f.typ)
```

Comment:
Style thing: you can just use `s = append(s, m.convert(...` in both branches of the condition. It would save one line.